comms/storage/logstream/logstream.go

```diff
@@ -109,53 +121,72 @@ func (ls LogStream) LogRebalanceEnd(prevShards []string, newShards []string) {
 }
 
 // GetNStatusUpdatesSince returns the first N status updates for a node that occurred on or after numHoursAgo hours ago.
-func (ls LogStream) GetNStatusUpdatesSince(n int, numHoursAgo float64, pubKey string) ([]NodeStatus, error) {
+func (ls LogStream) GetNStatusUpdatesSince(n int, numHoursAgo float64, pubKey string, logs *[]NodeStatus) error {
 	subject := fmt.Sprintf("storage.log.%s.statusUpdate", pubKey)
-	subscription, err := ls.jsc.SubscribeSync(
+	return GetNLogsForSubjSince(n, numHoursAgo, subject, ls.jsc, logs)
+}
+
+// GetNRebalanceStartSince returns the first N rebalanceStart logs for a node that occurred on or after numHoursAgo hours ago.
+func (ls LogStream) GetNRebalanceStartsSince(n int, numHoursAgo float64, pubKey string, logs *[]RebalanceStartLog) error {
+	subject := fmt.Sprintf("storage.log.%s.rebalanceStart", pubKey)
+	return GetNLogsForSubjSince(n, numHoursAgo, subject, ls.jsc, logs)
+}
+
+// GetNRebalanceEndsSince returns the first N rebalanceEnd logs for a node that occurred on or after numHoursAgo hours ago.
+func (ls LogStream) GetNRebalanceEndsSince(n int, numHoursAgo float64, pubKey string, logs *[]RebalanceEndLog) error {
+	subject := fmt.Sprintf("storage.log.%s.rebalanceEnd", pubKey)
+	return GetNLogsForSubjSince(n, numHoursAgo, subject, ls.jsc, logs)
+}
+
+// GetNUpdateHealthyNodeSetsSince returns the first N updateHealthyNodeSet logs for a node that occurred on or after numHoursAgo hours ago.
+func (ls LogStream) GetNUpdateHealthyNodeSetsSince(n int, numHoursAgo float64, logs *[]UpdateHealthyNodeSetLog) error {
+	return GetNLogsForSubjSince(n, numHoursAgo, "storage.log.updateHealthyNodeSet", ls.jsc, logs)
+}
+
+// GetNLogsForSubjSince sets *logs to the first N logs to subject starting from numHoursAgo hours ago.
+func GetNLogsForSubjSince[T any](n int, numHoursAgo float64, subject string, jsc nats.JetStreamContext, logs *[]T) error {
```

Comment:
I ended up making this generic to avoid copy/pasting very similar stream querying logic for each log type